src/main/java/org/broadinstitute/hellbender/engine/LocalReadShard.java

```diff
@@ -12,10 +11,7 @@
 
 import java.util.Iterator;
 import java.util.List;
-import java.util.Spliterator;
-import java.util.Spliterators;
 import java.util.stream.Collectors;
-import java.util.stream.StreamSupport;
 
 /**
  * A class to represent a shard of reads data, optionally expanded by a configurable amount of padded data.
```

Comment:
@droazen - looking into the resurrection of this branch, I realized that the `LocalReadShard` disappeared. For the sliding-window walker I required that each window is independent of the others, so I guess that the `MultiIntervalLocalReadShard` is not the option to go.

Is it ok to keep the `LocalReadShard`? Otherwise, I think that starting a new project for not keeping the GATK blow with unnecessary walkers for your methods is the best way to go...

Reply:
@magicDGS I think it would be ok to resurrect `LocalReadShard` if you really need it. But you should know that the reason we got rid of it was because we found that there are major performance issues with querying each interval individually. It's much more efficient to query a large number of intervals at once, particularly if the intervals are close together or overlap, to avoid reading and parsing the same parts of the bam file multiple times. Once we moved to `MultiIntervalLocalReadShard`, the performance of both the `HaplotypeCaller` and `Mutect2` increased by about 30-40%.

I'd recommend closing out these old PRs and submitting a new PR for `SlidingWindowWalker` with the walker class itself, an `ExampleSlidingWindowWalker` and `ExampleSlidingWindowWalker`, and whatever utilities you require (but note that the less common code you refactor, the easier it will be for us to merge your PR quickly).